jbpm-runtime-manager/src/main/java/org/jbpm/runtime/manager/impl/AbstractRuntimeManager.java

```diff
@@ -431,43 +448,31 @@ protected void createLockOnGetEngine(Context<?> context, RuntimeEngine runtime)
     }
     
     protected void createLockOnGetEngine(Long id, RuntimeEngine runtime) {
-        if (!isUseLocking()) {
-            logger.debug("Locking on runtime manager disabled");
+        if (id == null) {
             return;
         }
-        
-        if (id != null) {
-            ReentrantLock newLock = new ReentrantLock();
-            ReentrantLock lock = engineLocks.putIfAbsent(id, newLock);
-            if (lock == null) {
-                lock = newLock;
-                logger.debug("New lock created as it did not exist before");
-            } else {
-                logger.debug("Lock exists with {} waiting threads", lock.getQueueLength());
-            }
-            logger.debug("Trying to get a lock {} for {} by {}", lock, id, runtime);
-            lock.lock();
-            logger.debug("Lock {} taken for {} by {} for waiting threads by {}", lock, id, runtime, lock.hasQueuedThreads());
-            
+        try {
+            runtimeManagerLockStrategy.lock(id, runtime);
+        } catch(InterruptedException e) {
+            throw new RuntimeException("Runtime manager was not able to lock for {" + id +"}", e);
         }
-        
+    }
+
+    protected void releaseAndCleanLock(Long id, RuntimeEngine runtime) {
+        if (id == null) {
+            return;
+        }
+        runtimeManagerLockStrategy.unlock(id, runtime);
     }
     
     protected void createLockOnNewProcessInstance(Long id, RuntimeEngine runtime) {
         if (!isUseLocking()) {
```

Comment:
Check first if id is null (same as rest of methods)

Reply:
in this case the call is delegated to createLockOnGetEngine... so it is the same thing (it is checked in there.